android/src/main/java/com/reactnativeteadssdkmodule/ArrayUtil.kt

```diff
@@ -0,0 +1,113 @@
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+  The above copyright notice and this permission notice shall be included in all
+  copies or substantial portions of the Software.
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+  SOFTWARE.
+ */
+
+
+
+object ArrayUtil {
+  @Throws(JSONException::class)
+  fun toJSONArray(readableArray: ReadableArray): JSONArray {
+    val jsonArray = JSONArray()
+    for (i in 0 until readableArray.size()) {
+      val type = readableArray.getType(i)
+      when (type) {
+        ReadableType.Null -> jsonArray.put(i, null)
+        ReadableType.Boolean -> jsonArray.put(i, readableArray.getBoolean(i))
+        ReadableType.Number -> jsonArray.put(i, readableArray.getDouble(i))
+        ReadableType.String -> jsonArray.put(i, readableArray.getString(i))
+        ReadableType.Map -> jsonArray.put(i, MapUtil.toJSONObject(readableArray.getMap(i)))
+        ReadableType.Array -> jsonArray.put(i, toJSONArray(readableArray.getArray(i)))
+      }
+    }
+    return jsonArray
+  }
+
+  @Throws(JSONException::class)
+  fun toArray(jsonArray: JSONArray): Array<Any?> {
+    val array = arrayOfNulls<Any>(jsonArray.length())
+    for (i in 0 until jsonArray.length()) {
+      var value = jsonArray[i]
+      if (value is JSONObject) {
+        value = MapUtil.toMap(value)
+      }
+      if (value is JSONArray) {
+        value = toArray(value)
+      }
+      array[i] = value
+    }
+    return array
+  }
+
+  fun toArray(readableArray: ReadableArray): Array<Any?> {
+    val array = arrayOfNulls<Any>(readableArray.size())
+    for (i in 0 until readableArray.size()) {
+      val type = readableArray.getType(i)
+      when (type) {
+        ReadableType.Null -> array[i] = null
+        ReadableType.Boolean -> array[i] = readableArray.getBoolean(i)
+        ReadableType.Number -> array[i] = readableArray.getDouble(i)
+        ReadableType.String -> array[i] = readableArray.getString(i)
+        ReadableType.Map -> array[i] = MapUtil.toMap(readableArray.getMap(i))
+        ReadableType.Array -> array[i] = toArray(readableArray.getArray(i))
+      }
+    }
+    return array
+  }
+
+  fun toWritableArray(array: Array<Any?>?): WritableArray {
+    val writableArray = Arguments.createArray()
+    for (i in array!!.indices) {
```

Comment:
Arguments.createArray() never returns null writableArray?

Reply:
I think it can't return null :
<img width="607" alt="Capture d’écran 2023-04-11 à 10 34 16" src="https://user-images.githubusercontent.com/99195946/231103559-92519392-87cf-4757-bc89-61542eccbf67.png">